src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -215,12 +228,18 @@ def register_extra_cli_argument(command, dest, **kwargs):
     '''
     _cli_extra_argument_registry[command][dest] = CliCommandArgument(dest, **kwargs)
 
+def cli_command_with_handler(module_name, name, operation, client_factory=None, transform=None, table_transformer=None):
```

Comment:
I don't mean network caching. I mean why not save that command table (the bare bones now that basically maps the command names to their module) as a file on their system. If present it can just look up the command, and load that module etc instead of building everything. The mapping would only change for normal users if they updated their modules (which would be the obvious point to wipe that file if it exists) or in a dev scenario (where you could just disable caching entirely).

I do agree though, if rebuilding is "fast enough" then there's no point in doing the extra work. Of course, what is fast enough ;)